core/cap-0046-08.md

```diff
@@ -103,16 +80,15 @@ created contract.
 
 ### Topics
 In addition to the `SCVal data` in `ContractEvent`, we also provide an `SCVec`
-that the user can fill out with smaller values so downstream systems can
-provide more narrow filtering and subscription options without prior knowledge
-of the `data` format.
-
-TODO: Specify how topics will be limited in size.
+that the user can fill out with smaller values so downstream systems can provide
+more narrow filtering and subscription options without prior knowledge of the
+`data` format. There can only be four topics, and if the type is `Bytes`, cannot
+be more than 64 bytes.
```

Comment:
This 64 byte limit doesn't make much sense and should probably be removed. You can bypass it by using a Vec or String.

Reply:
I think it's probably worthwhile specifying the event limits more clearly. 
- Should the topics size be network configurable? `4` seems a bit arbitrary and too restrictive. 
- Should there be a size limit in both the topics and data? Unbounded events means contracts can submit a single event up to 500k (meta data size limit) repeatedly, causing downstream stress in processing them.